### PR TITLE
Add Random materializers for monad transformers

### LINKED ISSUE #1834: Add inductive materializations for Random
We should be able to materialize over monad transformers similarly to `Console`

### PR DESCRIPTION
**Pull request by alexandrustana**
This closes #1834

### CODE REVIEW

File: std/shared/src/main/scala/cats/effect/std/Random.scala [hunk 1]
```diff
@@ -140,6 +205,63 @@ object Random {
...
+   */
+  implicit def catsOptionTRandom[F[_]: Random: Functor]: Random[OptionT[F, *]] =
+    Random[F].mapK(OptionT.liftK)
+
+  /**
+   * [[Random]] instance built for `cats.data.StateT` values initialized with
+   * any `F` data type that also implements `Random`.
+   */
+  implicit def catsStateTRandom[F[_]: Random: Applicative, S]: Random[StateT[F, S, *]] =
+    Random[F].mapK(StateT.liftK)
```
Comment (djspiewak) [inline std/shared/src/main/scala/cats/effect/std/Random.scala]: This should probably be `IndexedStateT` just to be slightly more general

File: std/shared/src/main/scala/cats/effect/std/Random.scala [hunk 2]
```diff
@@ -140,6 +205,63 @@ object Random {
...
+   * `F` data type that also implements `Random`.
+   */
+  implicit def catsIorTRandom[F[_]: Random: Functor, L]: Random[IorT[F, L, *]] =
+    Random[F].mapK(IorT.liftK)
+
+  /**
+   * [[Random]] instance built for `cats.data.ReaderWriterStateT` values
+   * initialized with any `F` data type that also implements `Random`.
+   */
+  implicit def catsReaderWriterStateTRandom[
```
Comment (djspiewak) [inline std/shared/src/main/scala/cats/effect/std/Random.scala]: Also `Indexed`, ideally

**Review by djspiewak — CHANGES_REQUESTED** [review]
This is great! Just a couple super-minor changes. I'm thinking that the mima compatibility failures are spurious? Need to look at them a little closer and could use a second set of eyes.

**Comment by wemrysi** [comment]
@djspiewak I agree the mima issues are spurious, looks like the PR branch is based on a `series/3.x` pre #1854.

**Comment by alexandrustana** [comment]
@wemrysi you are right it's before #1854. I'll get the fix in when applying the requested changes

File: std/shared/src/main/scala/cats/effect/std/Random.scala (hunk 2 above)
Comment (alexandrustana) [reply std/shared/src/main/scala/cats/effect/std/Random.scala]: I wasn't aware of the Indexed version, should be ok now 🙂

**Pull request merged**